ConsoleApp3/RationalNumber.cs

```diff
@@ -0,0 +1,237 @@
+using System.Text;
+
+namespace ConsoleApp3
+{
+    class RationalNumber
+    {
+        private int numerator;              // Числитель
+        private int denominator;            // Знаменатель
+
+        public RationalNumber(int numerator, int denominator)
+        {
+            if (denominator == 0)
+            {
+                throw new DivideByZeroException("В знаменателе не может быть нуля");
+            }
+            this.numerator = numerator;
+            this.denominator = denominator;
+        }
+        public RationalNumber(int number)
+            : this(number, 1)
+        { }
+
+        public static bool operator ==(RationalNumber a, RationalNumber b)
+        {
+
+            Object aAsObj = a as Object;
+            Object bAsObj = b as Object;
+            if (aAsObj == null || bAsObj == null)
+            {
+                return aAsObj == bAsObj;
+            }
+            return a.Equals(b);
+        }
+
+        public static bool operator ==(RationalNumber a, int b)
+        {
+            return a == new RationalNumber(b);
+        }
+
+
+        public static bool operator ==(int a, RationalNumber b)
+        {
+            return new RationalNumber(a) == b;
+        }
+
+
+        public static bool operator !=(RationalNumber a, RationalNumber b)
+        {
+            return !(a == b);
+        }
+
+
+        public static bool operator !=(RationalNumber a, int b)
+        {
+            return a != new RationalNumber(b);
+        }
+
+
+        public static bool operator !=(int a, RationalNumber b)
+        {
+            return new RationalNumber(a) != b;
+        }
+
+
+        public override bool Equals(object obj)
+        {
+            if (!(obj is RationalNumber))
+            {
+                return false;
+            }
+
+            var number = (RationalNumber)obj;
+            return numerator == number.numerator &&
+                   denominator == number.denominator;
+        }
+
+        private static int getGreatestCommonDivisor(int a, int b)
+        {
+            while (b != 0)
+            {
+                int temp = b;
+                b = a % b;
+                a = temp;
+            }
+            return a;
+        }
+
+
+        public RationalNumber Reduce()
+        {
+            RationalNumber result = this;
+            int greatestCommonDivisor = getGreatestCommonDivisor(numerator, denominator);
+            result.numerator /= greatestCommonDivisor;
+            result.denominator /= greatestCommonDivisor;
+            return result;
+        }
+
+        private int CompareTo(RationalNumber that)
+        {
+            if (Equals(that))
+            {
+                return 0;
+            }
+            RationalNumber a = this.Reduce();
+            RationalNumber b = that.Reduce();
+            if (a.numerator * b.denominator > b.numerator * a.denominator)
+            {
+                return 1;
+            }
+            return -1;
```

Comment:
```suggestion
            return Comparer<double>.Default.Compare(a.numerator / (double)a.denominator, b.numerator / (double)b.denominator);
```
Превращаем рациональную дробь в вещественную и сравниваем стандартным сравнителем.